### PR TITLE
Add/remove custom controls dynamically

### PR DESCRIPTION
**Pull request by paodb**
~~This depends on the merge of https://github.com/FlowingCode/google-map/pull/35 and the release of the new version of the web-component.~~

This PR adds the possibility to add and remove custom controls dynamically. Also adds a method to remove all existing custom controls. Web component version was updated to 3.8.0.

## Release Notes

- **New Features**
	- Introduced a demo application showcasing custom controls for Google Maps, allowing dynamic addition and removal of controls.
	- Added a new CSS file for styling custom control buttons.

- **Enhancements**
	- Updated the Google Maps addon version to improve functionality and performance.
	- Enhanced management of custom controls within the Google Map component, including new methods for adding and removing controls.

- **Bug Fixes**
	- Resolved issues related to the handling of custom controls in the Google Map component.

### CODE REVIEW

**Comment by paodb** [comment]
@coderabbitai review

File: src/main/java/com/flowingcode/vaadin/addons/googlemaps/GoogleMap.java [hunk 1]
```diff
@@ -776,21 +780,52 @@
...
+      for (int i = 0; i < customControls.length; i++) {
+        CustomControl customControl = customControls[i];
+        jsonArray.set(i, customControl.getJson(i));
+        customControl.getControlButton().getElement().setAttribute("slot", "customControlSlot_" + i);
+        this.getElement().appendChild(customControl.getControlButton().getElement());
+        this.customControls.add(customControl);
+      }
+      this.getElement().setPropertyJson("customControls", jsonArray);
+    });    
+  }
```
[the remark this answers is not in the document]
Comment (paodb) [reply src/main/java/com/flowingcode/vaadin/addons/googlemaps/GoogleMap.java:802]: Fixed by https://github.com/FlowingCode/GoogleMapsAddon/pull/138/commits/4fa18e34525bfa4b61053dc105adf85fea23b3c8

File: src/test/java/com/flowingcode/vaadin/addons/googlemaps/CustomControlsDemo.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    resetButton.addClickListener(e -> {
+      gmaps.setCustomControls(customControl1, customControl2);
+      removeAllCustomControlsButton.setEnabled(true); // hide-source
+      addCustomControl3Button.setEnabled(true); // hide-source
+      removeCustomControl3Button.setEnabled(false); // hide-source
+    });
+   
+    add(new HorizontalLayout(addCustomControl3Button, removeCustomControl3Button,
+        removeAllCustomControlsButton, resetButton));
+  }
```
[the remark this answers is not in the document]
Comment (javier-godoy) [reply src/test/java/com/flowingcode/vaadin/addons/googlemaps/CustomControlsDemo.java:96]: They are just hidden lines in the demo...

File: src/main/java/com/flowingcode/vaadin/addons/googlemaps/GoogleMap.java [hunk 3]
```diff
@@ -42,19 +42,23 @@
...
-@NpmPackage(value = "@flowingcode/google-map", version = "3.7.1")
+@NpmPackage(value = "@flowingcode/google-map", version = "3.8.0")
 @NpmPackage(value = "@googlemaps/markerclusterer", version = "2.0.8")
 @JsModule("./googlemaps/geolocation.js")
 public class GoogleMap extends Component implements HasSize {
 
-  private Integer trackLocationId = null;
+  private Integer trackLocationId = null;  
+
+  private Set<CustomControl> customControls = new HashSet<CustomControl>();
```
[the remark this answers is not in the document]
Comment (paodb) [reply src/main/java/com/flowingcode/vaadin/addons/googlemaps/GoogleMap.java]: Was not sure about this one but yes, a List is better. Thanks. Fixed by https://github.com/FlowingCode/GoogleMapsAddon/pull/138/commits/4fa18e34525bfa4b61053dc105adf85fea23b3c8

**Review by javier-godoy — APPROVED** [review]

**Pull request merged**